src/components/layouts/auth-wrapper/auth-wrapper.js

```diff
@@ -73,6 +87,29 @@ class AuthWrapper extends React.PureComponent {
     shouldLogout: false
   };
 
+  componentDidMount() {
+    let token =
+      parse(this.props.location.search) &&
+      parse(this.props.location.search).token
+        ? parse(this.props.location.search).token
+        : undefined;
+
+    if (token) {
+      // eslint-disable-next-line no-undef
+      localStorage.setItem("token", token);
+
+      // eslint-disable-next-line no-undef
+      const lastUrlBeforeLogin = window.localStorage.getItem("last-visited");
+      if (lastUrlBeforeLogin) {
```

Comment:
This disallows the initial redirect to user edit

Reply:
One solution would be to parse a redirect instruction from the back and have a flag to give priority to that redirect instruction. What do you think?

Reply:
I'll add a query param `initial=true` so we can handle the verification that Chris requested. If that param exists then it will skip the redirect to lastUrl.